File: doc-examples/example-jsonb-java/src/main/java/example/ProgrammaticCodeSerializer.java
package example;

import jakarta.json.bind.serializer.JsonbSerializer;
import jakarta.json.bind.serializer.SerializationContext;
import jakarta.json.stream.JsonGenerator;

public final class ProgrammaticCodeSerializer implements JsonbSerializer<ProgrammaticCode> {
    @Override
    public void serialize(ProgrammaticCode obj, JsonGenerator generator, SerializationContext ctx) {
        generator.write("code:" + obj.getValue());
    }
}

File: doc-examples/example-jsonb-python/src/main/python/example/ProgrammaticCodeSerializer.py
from jakarta.json.bind.serializer import JsonbSerializer, SerializationContext
from jakarta.json.stream import JsonGenerator

from example.ProgrammaticCode import ProgrammaticCode


class ProgrammaticCodeSerializer(JsonbSerializer[ProgrammaticCode]):
    def serialize(self, obj: ProgrammaticCode, generator: JsonGenerator, ctx: SerializationContext) -> None:
        generator.write("code:" + obj.value)
